File: app/src/main/java/net/gtr/framework/rx/MessageDialog.java
/*
 * Copyright (c) 2017. heisenberg.gong
 */

package net.gtr.framework.rx;

import android.content.Context;
import android.view.View;

/**
 * Created by heisenberg on 2017/10/20.
 * heisenberg.gong@koolpos.com
 */

public interface MessageDialog {


    public void setCancelable(boolean mCancelable);

    public void setDialogMessage(CharSequence mMessage);

    public void setDialogTitle(CharSequence mTitle);

    public void setConfirmButtonText(CharSequence mConfirm);

    public void setConfirmButtonOnClickListener(View.OnClickListener onClickListener);

    public void setCancelButtonText(CharSequence mCancel);


    boolean isShowing();

    public void show();

    public void dismiss();

}

File: app/src/main/java/net/gtr/framework/rx/MessageDialog.kt
/*
 * Copyright (c) 2017. heisenberg.gong
 */

package net.gtr.framework.rx

import android.content.Context
import android.view.View

/**
 * Created by heisenberg on 2017/10/20.
 * heisenberg.gong@koolpos.com
 */

interface MessageDialog {


    val isDialogShowing: Boolean

    fun setCancelable(mCancelable: Boolean)

    fun setDialogMessage(mMessage: CharSequence)

    fun setDialogTitle(mTitle: CharSequence)

    fun setConfirmButtonText(mConfirm: CharSequence)

    fun setConfirmButtonOnClickListener(onClickListener: View.OnClickListener)

    fun setCancelButtonText(mCancel: CharSequence)

    fun show()

    fun dismiss()

}
